Data/ByteString/Utils/ByteOrder.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE CPP #-}
+
+-- | Why does this module exist? There is "GHC.ByteOrder" in base.
+-- But that module is /broken/ until base-4.14/ghc-8.10, so we
+-- can't rely on it until we drop support for older ghcs.
+-- See https://gitlab.haskell.org/ghc/ghc/-/issues/20338
+-- and https://gitlab.haskell.org/ghc/ghc/-/issues/18445
+
+#include "MachDeps.h"
+
+module Data.ByteString.Utils.ByteOrder
+  ( ByteOrder(..)
+  , hostByteOrder
```

Comment:
Is the name intentionally different from `targetByteOrder` for cross-compilation purposes?

Reply:
Since `bytestring` is not a compiler the notion of "target" doesn't really make sense. The name in `base` is confusing.